application/comit_node/src/http_api/rfc003/action.rs

```diff
@@ -105,6 +105,30 @@ impl FromAcceptSwapRequestHttpBody<Bitcoin, Ethereum>
     }
 }
 
+impl FromAcceptSwapRequestHttpBody<Ethereum, Bitcoin>
+    for StateMachineResponse<
+        ethereum_support::Address,
+        secp256k1_support::KeyPair,
+        bitcoin_support::Blocks,
+    >
+{
+    fn from_accept_swap_request_http_body(
+        body: AcceptSwapRequestHttpBody<Ethereum, Bitcoin>,
+        id: SwapId,
+        secret_source: &dyn SecretSource,
+    ) -> Result<Self, HttpApiProblem> {
+        match body {
+            AcceptSwapRequestHttpBody::OnlyRefund { .. } | AcceptSwapRequestHttpBody::RefundAndRedeem { .. } => Err(HttpApiProblem::with_title_and_type_from_status(400).set_detail("The refund identity for swaps where Bitcoin is the BetaLedger has to be provided on-demand, i.e. when the refund action is executed.")),
+            AcceptSwapRequestHttpBody::None { .. } => Err(HttpApiProblem::with_title_and_type_from_status(400).set_detail("Missing beta_ledger_redeem_identity")),
+            AcceptSwapRequestHttpBody::OnlyRedeem { alpha_ledger_redeem_identity, beta_ledger_lock_duration } => Ok(StateMachineResponse {
```

Comment:
❓ we use the `OnlyRedeem` body for http swap accept request? This is weird, can't we use a dedicated body type?

Reply:
Dedicated to what in particular? Isn't `OnlyRedeem` a dedicated body type?

Reply:
I think I just don't like the naming of `AcceptSwapRequestHttpBody` enum.

Now I understand.

I can't think of something clearer. Maybe `OnlyRedeemInfo` but it would make it repeated.

I'll think about it. All good for this PR.